gpii/node_modules/dbOperation/test/DbDataStoreTestsUtils.js

```diff
@@ -0,0 +1,326 @@
+                    "%dbOperation/test/data/gpiiKeys.json",
+                    "%dbOperation/test/data/prefsSafes.json",
+                    "%gpii-universal/testData/dbData/views.json"
+                ]
+            }
+        }
+    },
+    components: {
+        testCaseHolder: {
+            type: "gpii.tests.dbDataStore.baseTestCaseHolder"
+        }
+    },
+    distributeOptions: {
+        source: "{that}.options.rawModules",
+        target: "{that > testCaseHolder}.options.rawModules"
+    },
+    mergePolicy: {
+        rawModules: "noexpand"
+    }
+});
+
+fluid.defaults("gpii.tests.dbDataStore.baseTestCaseHolder", {
+    gradeNames: ["gpii.test.express.caseHolder"],
+    events: {
+        onResponse: null,
+        onError: null
+    },
+    components: {
+        dbDataStore: {
+            type: "gpii.dbOperation.dbDataStore",
+            options: {
+                dataSourceConfig: {
+                    baseUrl: "http://localhost",
+                    port: "{gpii.tests.dbDataStore.environment}.options.port",
+                    dbName: "gpii"
+                }
+            }
+        }
+    },
+    sequenceEnd: [{
+        func: "{gpii.tests.dbDataStore.environment}.events.onCleanup.fire"
+    }, {
+        event:    "{gpii.tests.dbDataStore.environment}.events.onCleanupComplete",
+        listener: "fluid.log",
+        args:     ["Database cleanup complete"]
+    }]
+});
+
+gpii.tests.dbDataStore.saveAndInvokeFetch = function (fetchDataSource, id, that) {
+    gpii.tests.dbDataStore.lastSavedId = id;
+    gpii.tests.dbOperation.invokePromiseProducer(fetchDataSource, [id], that);
+};
+
+gpii.tests.dbDataStore.verifyFetched = function (response, expected) {
+    jqUnit.assertEquals("The fetched document id matches the saved record", gpii.tests.dbDataStore.lastSavedId, response.id);
+    jqUnit.assertLeftHand("The data is saved successfully", expected, response);
+};
+
+gpii.tests.dbDataStore.verifyFetchedGpiiKey = function (response, expected) {
+    // verify the GPII key
+    var gpiiKey = expected.gpiiKey;
+    if (gpiiKey) {
+        jqUnit.assertEquals("The \"id\" value has been set to the provided key", gpiiKey, response.id);
+    } else {
+        jqUnit.assertNotUndefined("The \"id\" value has been set to an auto generated key", response.id);
+    }
+
+    // verify other fields
+    var nonIdFields = fluid.censorKeys(expected, ["gpiiKey"]);
+    gpii.tests.dbDataStore.verifyFetched(response, nonIdFields);
+    jqUnit.assertEquals("The \"type\" value has been set to default", gpii.dbOperation.docTypes.gpiiKey, response.type);
+    jqUnit.assertEquals("The \"schemaVersion\" value has been set to default", gpii.dbOperation.schemaVersion, response.schemaVersion);
+    jqUnit.assertEquals("The \"prefsSafeId\" value has been set to default", nonIdFields.prefsSafeId, response.prefsSafeId);
+    jqUnit.assertEquals("The \"prefsSetId\" value has been set to default", nonIdFields.prefsSetId, response.prefsSetId);
```

Comment:
I think the messages here for `prefsSafeId` and `prefsSetId` should read "has been set to the expected value". Rather than "default".

Though, aren't these 2 tests taken care of by the line:

```
gpii.tests.dbDataStore.verifyFetched(response, nonIdFields);
```